determined-ai/determined
# feat: Convert userRoles to observables, fix logout issue [WEB-798]

## Description

Converts userRoles and userAssignments store from Loadable to micro-observables

Related to WEB-732 - logging out from a user will clear the internal memory of its userAssignments and userRoles

Where are the affected functions used?
- src/components/DeterminedAuth.tsx
- src/components/Navigation.tsx
- src/hooks/usePermissions.ts
- [new] src/pages/SignOut.tsx

## Test Plan

Test permissions and user data work correctly in EE environment:

- Sign in as admin.
- Confirm you can see workspaces, and admin user badge in top left nav.
- Find a workspace. In its members tab, give 'determined' a Viewer role.
- Sign out, then sign back in as determined.
- Confirm determined can see this workspace and members (but not edit / add members)

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
